spec/cache_store_redis_spec.rb

```diff
@@ -1,198 +1,227 @@
 
-      expect(v.class).to eq(TestObject)
-      expect(v.text).to eq('abc123')
-      expect(v.numeric).to eq(123)
+    subject { described_class.new('myname') }
 
-      v2 = @cache_store.get(key)
-      expect(v2.class).to eq(TestObject)
-      expect(v2.text).to eq('abc123')
-      expect(v2.numeric).to eq(123)
+    it 'sets the value' do
+      subject.public_send(method_name, key, value)
+
+      expect(subject.get(key)).to eq(value)
     end
 
-    context 'when set value has expired' do
-      let(:key  ) { SecureRandom.uuid }
-      let(:value) { '123'             }
+    context 'when key already exists' do
+      let(:new_value) { 'Peter' }
+      before { subject.public_send(method_name, key, value) }
 
-      before :each do
-        @cache_store.set(key, value, 1)
-        sleep(1.2)  # TODO: find a way to remove this by stubbing the Redis expire mechanism
-      end
+      it 'updates the item' do
+        subject.public_send(method_name, key, new_value)
 
-      it 'returns nil' do
-        expect(@cache_store.get(key)).to be_nil
+        expect(subject.get(key)).to eq(new_value)
       end
     end
+  end
+end
 
-    context 'when namespaced' do
-      let(:key  ) { 'name' }
-      let(:value) { 'Tom'  }
-
-      subject { described_class.new('myname') }
-
-      it 'sets the value' do
-        @cache_store.set(key, value)
+RSpec.shared_examples '#get' do |method_name|
+  let(:value) { 'value' }
+  let(:key) { 'getkey' }
 
-        expect(@cache_store.get(key)).to eq(value)
-      end
+  it 'runs the hyrdation block when the value is not in the cache' do
+    v = subject.public_send(method_name, key) do
+      value
+    end
 
-      context 'when key already exists' do
-        let(:new_value) { 'Peter' }
-        before { @cache_store.set(key, value) }
+    expect(subject.public_send(method_name, key)).to eq v
+  end
 
-        it 'updates the item' do
-          @cache_store.set(key, new_value)
+  context 'when the value in the store is empty string' do
+    let(:value) { '' }
 
-          expect(@cache_store.get(key)).to eq(new_value)
-        end
-      end
+    it 'does not attempt to deserialize' do
+      subject.set(key, value)
+      expect(subject).to_not receive(:deserialize)
+      expect(subject.public_send(method_name, key)).to be nil
     end
   end
+end
 
-  describe '#get' do
-    let(:value) { 'value' }
-    let(:key) { 'getkey' }
+RSpec.shared_examples '#remove' do |method_name|
+  context 'when the value exists' do
+    let(:key  ) { SecureRandom.uuid }
+    let(:value) { '123'             }
 
-    it 'runs the hyrdation block when the value is not in the cache' do
-      v = @cache_store.get(key) do
-        value
-      end
+    before { subject.set(key, value) }
+
+    it 'removes that value' do
+      expect(subject.exist?(key)).to eq(true)
 
-      expect(@cache_store.get(key)).to eq v
+      subject.public_send(method_name, key)
+
+      expect(subject.exist?(key)).to eq(false)
```

Comment:
[minor] `.to be false`

Reply:
These were existing specs, I've just moved them without changing them.

Reply:
Updated in [fe053c1](https://github.com/Sage/cache_store_redis/pull/13/commits/fe053c1fc173616919eeaef447e81b5c447aef70)